source/s3_auto_ranged_put.c

```diff
@@ -330,24 +741,30 @@ static int s_s3_auto_ranged_put_prepare_request(
         }
         case AWS_S3_AUTO_RANGED_PUT_REQUEST_TAG_PART: {
 
-            size_t request_body_size = meta_request->part_size;
+            size_t request_body_size = s_compute_request_body_size(meta_request, request->part_number);
 
-            /* Last part--adjust size to match remaining content length. */
-            if (request->part_number == auto_ranged_put->synced_data.total_num_parts) {
-                size_t content_remainder =
-                    (size_t)(auto_ranged_put->content_length % (uint64_t)meta_request->part_size);
-
-                if (content_remainder > 0) {
-                    request_body_size = content_remainder;
+            if (request->num_times_prepared == 0) {
+                if (s_skip_parts_from_stream(
+                        meta_request,
+                        auto_ranged_put->prepare_data.num_parts_read_from_stream,
+                        request->part_number - 1)) {
+                    /* BEGIN CRITICAL SECTION */
+                    {
+                        aws_s3_meta_request_lock_synced_data(meta_request);
+                        aws_s3_meta_request_set_fail_synced(meta_request, NULL, AWS_ERROR_S3_RESUME_FAILED);
+                        aws_s3_meta_request_unlock_synced_data(meta_request);
+                    }
```

Comment:
seems like from `s_s3_meta_request_prepare_request_task` function, as long as you raised the error code correctly and return NULL; it will call `aws_s3_meta_request_set_fail_synced`.  [here](https://github.com/awslabs/aws-c-s3/blob/main/source/s3_meta_request.c#L558-L578).

Why not follow the previous way of error handling?